### PR TITLE
IDAT entry_point.

### PR DESCRIPTION
**Pull request by rajwanir**
This PR adds IDAT entry_point and makes the workflow truly raw data to QC report.

1.  The IDAT entry point uses a Illumina's free Dragena software (https://help.dragenarray.illumina.com/product-guides/dragen-array-local-analysis) for converting IDAT to GTC (383b89bd5225808973a47c9357136d8339b60c0e). Since Dragena is Illumina's proprietary software and unavailable via Conda, the pipliene can only use it as an enviornment module.
2. Non-CCAD2 users can download the dragen array from Illumina's website and provide a path to dragena binary via the `dragena_location` option in config to use the IDAT entry_point (see 55655f629fa30a8a35918fa657636a334973353b). 
3. After IDAT to GTC conversion, the workflow resumes GTC entry point with either BCF or PED option. (13cf63d21eb2f996b977bb345f0b38ee4b493ec5)
4. The IDAT entry point can run both locally and in cluster_mode (13cf63d21eb2f996b977bb345f0b38ee4b493ec5)

I can update the docs once PR is finalized. Thank you.

### CODE REVIEW

**Comment by jaamarks** [comment]
Concept ACK - will test.

File: src/cgr_gwas_qc/workflow/modules/idat.smk [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+rule idat2gtc:
+    input:
+        idat_ss=rules.write_idat2gtc_ss.output[0],
+    params:
+        bpm=cfg.config.reference_files.illumina_manifest_file,
+        egt=cfg.config.reference_files.illumina_cluster_file,
+        dragena_location=cfg.config.workflow_params.dragena_location,
+    output:
+        output_folder=directory("sample_level/gtcs"),
+    threads: 44
```
Comment (jaamarks) [inline src/cgr_gwas_qc/workflow/modules/idat.smk:46]: This could benefit from your approach in PR #361

File: src/cgr_gwas_qc/workflow/modules/idat.smk [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    envmodules:
+        "dragena/1.0.0",
+    shell:
+        """
+        if [ "{params.dragena_location}" != "None" ];then dragena='{params.dragena_location}';else  dragena='dragena';fi
+        $dragena genotype call --bpm-manifest {params.bpm} --cluster-file {params.egt} --idat-sample-sheet {input.idat_ss} --num-threads {threads} --output-folder {output.output_folder}
```
Comment (jaamarks) [inline src/cgr_gwas_qc/workflow/modules/idat.smk]: I think we could add another check to handle the case when dragena is not available. For example wrong path to executable.
```

if ! command -v $dragena &> /dev/null; then
    echo "Error: Dragena executable not found. Ensure that 'dragena/1.0.0' module is loaded or provide a valid 'dragena_location' in the config." >&2
    exit 1
fi

```

**Review by jaamarks — COMMENTED** [review]
I generally agree with these changes. The will enhance the generality of the pipeline and effectively address several edge cases for specific chips. 

I have a series of tests planned, but I'm currently waiting in the queue for CCAD2. I will update you with the results once the tests are complete.

Nice work 👍

**Review by jaamarks — CHANGES_REQUESTED** [review]
I noticed that in `cgr_gwas_qc/models/config/reference_files.py` the docstring for ReferenceFiles class is missing `illumina_cluster_file`. This affects the docs. Can you add this into your PR?

**Comment by rajwanir** [comment]
> I noticed that in `cgr_gwas_qc/models/config/reference_files.py` the docstring for ReferenceFiles class is missing `illumina_cluster_file`. This affects the docs. Can you add this into your PR?

Added in d25604d13cad6c4cde6971c47bcae14b25b12ff1.

File: src/cgr_gwas_qc/workflow/modules/idat.smk (hunk 2 above)
Comment (rajwanir) [reply src/cgr_gwas_qc/workflow/modules/idat.smk]: Thanks for the suggestion.

Added in 3e4469ff40c593b3f818c0415f59b9821308bc83

File: src/cgr_gwas_qc/workflow/modules/idat.smk (hunk 1 above)
Comment (rajwanir) [reply src/cgr_gwas_qc/workflow/modules/idat.smk:46]: Agree, it would. But PR #361 is yet to be merged. How about we merge this and add a commit later to replace `threads: 44` with `threads: workflow.cores` once  PR #361 is merged. Thanks

**Review by jaamarks — APPROVED** [review]
Everything looks good! We tested this PR using a sample size of N=500, and the summary tables matched the results from previous runs that did not include this feature. 

After reviewing all aspects and confirming the successful execution of the GitHub Actions test_and_build job, I approve this PR.
